### PR TITLE
24 create trip app

### PR DESCRIPTION
**Pull request by PanLeon**
## TRELLO

* [Main Trello ticket](https://trello.com/b/BJDdC443/mytrip)

## Code reviewers

- [x] @Ashiato
- [ ] @black-art
- [x] @hrytskivr

### Second Level Review

- [ ] @byshovets
- [ ] @lhalam

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: myTrip/trip/views.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        else:
+            trips = Trip.get_all()
+            trips = [trip.to_dict() for trip in trips]
+            return JsonResponse(trips, status=200, safe=False)
+
+    def post(self, request):
+        """Handles POST request."""
+        data = json.loads(request.body.decode('utf-8'))
+        Trip.create(**data)
+        return HttpResponse(status=200)
```
Comment (oshtohryn) [inline myTrip/trip/views.py]: need to return status 201 Created

File: myTrip/trip/urls.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from django.conf.urls import url
+from . import views
+
+urlpatterns = [
+    url(r'^$',views.TripView.as_view()),
+    url(r'^(?P<trip_id>\d+)/$', views.TripView.as_view()),
+]
```
Comment (oshtohryn) [inline myTrip/trip/urls.py]: No newline at end of file

File: myTrip/trip/models.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+     :argument id: int - auto generated primary key
+     :argument user_id: int - ToDo foreign key to User model
+     :argument title: str - title
+     :argument describtion: text - describtion
+     :argument created_at: date - date
+     :argument status: int - 0-in progres, 1-annonced, 2-finished
+    ."""
+    user_id = models.IntegerField()
+    title = models.CharField(max_length=200)
+    describtion = models.TextField()
```
Comment (oshtohryn) [inline myTrip/trip/models.py]: descriBtion should be description

File: myTrip/trip/models.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+from django.db import models
+from django.core.exceptions import ObjectDoesNotExist
+
+class Trip(models.Model):
+    """
+     Trip
+     :argument id: int - auto generated primary key
+     :argument user_id: int - ToDo foreign key to User model
+     :argument title: str - title
+     :argument describtion: text - describtion
```
Comment (oshtohryn) [inline myTrip/trip/models.py]: description

File: myTrip/trip/models.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    @staticmethod
+    def get_by_id(trip_id):
+        """method to get trip by id"""
+        try:
+            return Trip.objects.get(id=trip_id)
+        except ObjectDoesNotExist:
+            return None
+
+    @staticmethod
+    def create(**data):
```
Comment (oshtohryn) [inline myTrip/trip/models.py]: only 1 time need to use **data

File: myTrip/trip/apps.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class CreateTripConfig(AppConfig):
+    name = 'trip'
```
Comment (hrytskivr) [inline myTrip/trip/apps.py:5]: Suggestion: add docstring. Godknowswhy :D

File: myTrip/trip/models.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        trip = Trip(**data)
+        trip.save()
+        return None
+
+    @staticmethod
+    def edit(data, trip_id):
+        """method to update trip fields (title,describtion,status)"""
+        trip = Trip.objects.get(id=trip_id)
+        trip.title = data['title']
+        trip.describtion = data['describtion']
```
Comment (hrytskivr) [inline myTrip/trip/models.py]: description :)

File: myTrip/trip/apps.py (hunk 6 above)
Comment (lhalam) [reply myTrip/trip/apps.py:5]: pylint ignore=CVS, migrations, apps.py, urls.py, wsgi.py. local_setings.py, manage.py

File: myTrip/trip/models.py [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+     :argument created_at: date - date
+     :argument status: int - 0-in progres, 1-annonced, 2-finished
+    ."""
+    user_id = models.IntegerField()
+    title = models.CharField(max_length=200)
+    description = models.TextField()
+    created_at = models.DateTimeField(default=datetime.now, blank=True)
+    status = models.IntegerField(default=0)
+
+    def to_dict(self):
```
Comment (lhalam) [inline myTrip/trip/models.py:23]: for example:
```
"""Convert model object to dictionary.
        Return:
            dict:
                {
                    'id': id,
                    'user_id': name,
                    ...
                }
        """
```

File: myTrip/trip/models.py [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        return {
+            "id":self.id,
+            "user":self.user_id,
+            "title": self.title,
+            "created_at": self.created_at,
+            "description": self.description,
+            "status": self.status}
+
+    def __str__(self):
+        return self.id
```
Comment (lhalam) [inline myTrip/trip/models.py]: add other args

File: myTrip/trip/models.py [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+    @classmethod
+    def get_all(cls):
```
Comment (lhalam) [inline myTrip/trip/models.py]: don't use the method all

File: myTrip/trip/models.py [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+        """method to update trip fields (title,description,status)"""
+        trip = Trip.objects.get(id=trip_id)
+        trip.title = data['title']
+        trip.description = data['description']
+        trip.status = data['status']
+        trip.save()
+        return None
+
+    @staticmethod
+    def delete_trip(trip_id):
```
Comment (lhalam) [inline myTrip/trip/models.py]: rename delete_by_id

File: myTrip/trip/models.py [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+        return self.id
+
+    @classmethod
+    def get_all(cls):
+        """method to get all trips"""
+        return Trip.objects.all()
+
+    @staticmethod
+    def get_by_id(trip_id):
+        """method to get trip by id"""
```
Comment (lhalam) [inline myTrip/trip/models.py]: for example:
```
"""
Get Like with given trip id
 Args:
    trip_id (int): trip id.
Returns:
    QuerySet<Like>: QuerySet of Like.
"""
```

File: myTrip/trip/models.py [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+        try:
+            return Trip.objects.get(id=trip_id)
```
Comment (lhalam) [inline myTrip/trip/models.py]: create instance trip
trip = Trip.objects.get(id=trip_id)
return trip

File: myTrip/trip/models.py [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+        except ObjectDoesNotExist:
+            return None
+
+    @staticmethod
+    def create(data):
+        """method to create trip"""
+        trip = Trip(**data)
+        trip.save()
+        return None
```
Comment (lhalam) [inline myTrip/trip/models.py]: return created the instance

File: myTrip/trip/models.py [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+
+    @staticmethod
```
Comment (lhalam) [inline myTrip/trip/models.py]: don't use @staticmethod for methods update, delete,

**Review by yuriiZherebukh — APPROVED** [review]

**Pull request merged**